feat: adding `stats/incr/nanmmse`

satansin123 opened the pull request:
---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes. report:
  - task: lint_filenames status: passed
  - task: lint_editorconfig status: passed
  - task: lint_markdown status: passed
  - task: lint_package_json status: passed
  - task: lint_repl_help status: passed
  - task: lint_javascript_src status: passed
  - task: lint_javascript_cli status: na
  - task: lint_javascript_examples status: passed
  - task: lint_javascript_tests status: passed
  - task: lint_javascript_benchmarks status: passed
  - task: lint_python status: na
  - task: lint_r status: na
  - task: lint_c_src status: na
  - task: lint_c_examples status: na
  - task: lint_c_benchmarks status: na
  - task: lint_c_tests_fixtures status: na
  - task: lint_shell status: na
  - task: lint_typescript_declarations status: passed
  - task: lint_typescript_tests status: passed
  - task: lint_license_headers status: passed ---

Resolves #5599.

## Description

> What is the purpose of this pull request?

This pull request:

-   Adds the @stdlib/stats/incr/nanmmse package, which computes an incremental mean while handling NaN values by skipping them.

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #5599

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

test - 
![image](https://github.com/user-attachments/assets/432717bf-f32b-4303-be5f-0ad27b0a8813)

examples -
![image](https://github.com/user-attachments/assets/c5083ee6-7ea2-4a35-b9a2-f97c53761530)

benchmark - 
![image](https://github.com/user-attachments/assets/e56cdb0e-7ddb-4cf9-bf54-356262dd2a26)


## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

satansin123 committed:
feat: adding stats/incr/nanmmse

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: passed
  - task: lint_repl_help
    status: passed
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: passed
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: passed
  - task: lint_license_headers
    status: passed
---

hrshya commented on lib/node_modules/@stdlib/stats/incr/nanmmse/README.md:
```diff
@@ -0,0 +1,181 @@
+
+#### incrnanmmse( window )
+
+Returns an accumulator `function` which incrementally computes a moving [mean squared error][mean-squared-error]. The `window` parameter defines the number of values over which to compute the moving [mean squared error][mean-squared-error].
```
```suggestion
Returns an accumulator `function` which incrementally computes a moving [mean squared error][mean-squared-error]. The `window` parameter defines the number of values over which to compute the moving [mean squared error][mean-squared-error], ignoring `NaN` values.
```

hrshya commented on lib/node_modules/@stdlib/stats/incr/nanmmse/lib/main.js:
```diff
@@ -0,0 +1,94 @@
+* m = accumulator();
+* // returns 33.0
+*/
+function incrnanmmse( W ) {
```
We need to create a thin wrapper around the `@stdlib/stats/incr/mmse`. So there is no need to implement it from scratch.

satansin123 replied:
hi @hrshya, i have styled as per your request, please review.

hrshya reviewed (commented):
The Implementation is wrong what you need to do is create a thin wrapper around the `stats/incr/mmse`.

satansin123 committed:
style: as suggested by the reviewer

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

hrshya commented on lib/node_modules/@stdlib/stats/incr/nanmmse/lib/main.js:
```diff
@@ -21,7 +21,7 @@
 var isPositiveInteger = require( '@stdlib/assert/is-positive-integer' ).isPrimitive;
-var incrmmean = require( '@stdlib/stats/incr/mmean' );
+var incrmmse = require( '@stdlib/stats/incr/mmse' );
 var format = require( '@stdlib/string/format' );
```
This module is not being used so you can remove it.

hrshya commented on lib/node_modules/@stdlib/stats/incr/nanmmse/lib/main.js:
```diff
@@ -63,11 +63,11 @@ var isnan = require( '@stdlib/math/base/assert/is-nan' );
 function incrnanmmse( W ) {
-	var mean;
+	var mmse;
 	if ( !isPositiveInteger( W ) ) {
```
This condition is being handled in `mmse` so there is no need to check it here.

satansin123 replied:
@hrshya its also done, thanks for the review

hrshya reviewed (commented):
Just some minor changes. Rest, the PR looks good!

satansin123 committed:
style: as suggested by the reviewer

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

hrshya commented:
Great work! Time to wait for a maintainer to merge this PR :)

Planeshifter commented on lib/node_modules/@stdlib/stats/incr/nanmmse/package.json:
```diff
@@ -0,0 +1,84 @@
+{
+  "name": "@stdlib/stats/incr/nanmmse",
+  "version": "0.0.0",
+  "description": "Compute a moving mean squared error (MSE) incrementally, ignoring `NaN` values.",
```
Use plain `NaN` in the description, not backtick-escaped.

```suggestion
  "description": "Compute a moving mean squared error (MSE) incrementally, ignoring NaN values.",
```

Planeshifter committed:
docs: update description

Co-authored-by: Harsh <149176984+hrshya@users.noreply.github.com>
Signed-off-by: Philipp Burckhardt <pburckhardt@outlook.com>

Planeshifter committed:
docs: remove backticks

Signed-off-by: Philipp Burckhardt <pburckhardt@outlook.com>

Planeshifter commented on lib/node_modules/@stdlib/stats/incr/nanmmse/README.md:
```diff
@@ -0,0 +1,181 @@
+
+-   <span class="package-name">[`@stdlib/stats/incr/mrmse`][@stdlib/stats/incr/mrmse]</span><span class="delimiter">: </span><span class="description">compute a moving root mean squared error (RMSE) incrementally.</span>
+-   <span class="package-name">[`@stdlib/stats/incr/mrss`][@stdlib/stats/incr/mrss]</span><span class="delimiter">: </span><span class="description">compute a moving residual sum of squares (RSS) incrementally.</span>
+-   <span class="package-name">[`@stdlib/stats/incr/mse`][@stdlib/stats/incr/mse]</span><span class="delimiter">: </span><span class="description">compute the mean squared error (MSE) incrementally.</span>
```
```suggestion
```

Planeshifter committed:
docs: make related packages section empty

Signed-off-by: Philipp Burckhardt <pburckhardt@outlook.com>

Planeshifter commented on lib/node_modules/@stdlib/stats/incr/nanmmse/README.md:
```diff
@@ -0,0 +1,174 @@
+<!-- Section for related `stdlib` packages. Do not manually edit this section, as it is automatically populated. -->
+
+<section class="related">
+
+
+</section>
```
```suggestion
<section class="related">

</section>
```

Planeshifter committed:
docs: remove empty line

Signed-off-by: Philipp Burckhardt <pburckhardt@outlook.com>

Planeshifter commented on lib/node_modules/@stdlib/stats/incr/nanmmse/README.md:
```diff
@@ -0,0 +1,173 @@
+[@stdlib/stats/incr/mrss]: https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/stats/incr/mrss
+
+[@stdlib/stats/incr/mse]: https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/stats/incr/mse
+
```
```suggestion
```

Planeshifter committed:
docs: remove links

Signed-off-by: Philipp Burckhardt <pburckhardt@outlook.com>

Planeshifter reviewed (dismissed):
Let's land this PR! hanks @satansin123 and @hrshya for review!

Planeshifter commented:
/stdlib update-copyright-years

Planeshifter commented:
/stdlib merge

stdlib-bot committed:
Merge remote-tracking branch 'upstream/develop' into feature/adding-nanmmse-package-5599

Planeshifter commented:
/stdlib update-copyright-years

stdlib-bot committed:
chore: update copyright years
